lib/arclight/normalized_date.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+module Arclight
+  ##
+  # A utility class to normalize dates, typically by joining
+  # e.g., "1990-2000, bulk 1990-1999"
+  # @see http://www2.archivists.org/standards/DACS/part_I/chapter_2/4_date
+  class NormalizedDate
+    # @param [String | Array<String>] `inclusive` from the `unitdate`
+    # @param [String] `bulk` from the `unitdate`
+    # @param [String] `other` from the `unitdate` when type is not specified
+    def initialize(inclusive, bulk = nil, other = nil)
+      if inclusive.is_a? Array # of YYYY-YYYY for ranges
+        @inclusive = YearRange.new(inclusive.include?('/') ? inclusive : inclusive.map { |v| v.tr('-', '/') }).to_s
+      elsif inclusive.present?
+        @inclusive = inclusive.strip
+      end
+      @bulk = bulk.strip if bulk.present?
+      @other = other.strip if other.present?
+    end
+
+    # @return [String] the normalized title/date
+    def to_s
+      normalize
+    end
+
+    private
+
+    attr_reader :inclusive, :bulk, :other
+
+    # @see http://www2.archivists.org/standards/DACS/part_I/chapter_2/4_date for rules
+    def normalize
+      if inclusive.present?
+        result = inclusive.to_s
+        result << ", bulk #{bulk}" if bulk.present?
+      elsif other.present?
+        result = other.to_s
+      else
+        result = nil
+      end
+      return if result.blank?
+      filter_dates(result)
+    end
+
+    def filter_dates(dates)
+      dates.gsub!('n.d.', 'undated ') # use "undated" instead of "n.d."
+      dates.gsub!('c.', 'circa ') # use "circa" instead of "c."
```

Comment:
Per [#319 (comment)](https://github.com/sul-dlss/arclight/issues/319#issuecomment-301969809), my opinion is that we should not be normalizing them here - we should just accept what exists in the text node already. I think this kind of normalization is out of scope for us.